src/models/mst.test.ts

```diff
@@ -742,30 +764,35 @@ describe("mst", () => {
 
       expect(itemList.items.length).toBe(1);
 
-      expect(output).toEqual([
-        "one: description of one",
-        "two: description of two",
-        "one: undefined"
+      expect(log).toEqual([
+        "computedValue called on one",
+        "autorun: one - one: undefined"
       ]);
 
       // Clean up
       disposers.forEach((disposer) => disposer());
     });
 
     test("with isAlive no warning is printed", () => {
-      const { itemList, output } = initialize();
+      const { itemList, log, clearLog } = initialize();
 
       // This list of autorun reactions emulates a list of MobX React
-      // that are monitoring the items in this list
-      const disposers = itemList.items.map((item) => autorun(() => {
-        const line = isAlive(item) ? item.nameWithDescription : "not alive";
-        output.push(line);
-      }));
+      // components that are monitoring the items in this list.
+      // However with a real component the render call is not run
+      // directly. Instead the need to re-render is queued up when
+      // a dependency changes. If a parent of the component stops
+      // rendering this component in the meantime the re-render will
+      // never happen.
+      const disposers = itemList.items.map((item) => {
+        const itemName = item.name;
+        return autorun(() => {
+          const line = isAlive(item) ? item.nameWithDescription : "not alive";
```

Comment:
I've removed this style in favor of a short circuit approach. With the short circuit it isn't necessary to figure out what to return when the item is not alive. And it also means the short circuit can be added at the beginning of a the render function instead of buried down where a property is accessed.